examples/models/data_tables_server.py

```diff
@@ -23,10 +27,14 @@ def __init__(self):
         self.class_filter = None
 
         self.source = ColumnDataSource()
+        # A bit hacky, but now we have an index column that will carry through our slicing
+        self.df = DataFrame.from_dict(ColumnDataSource.from_df(mpg))
+        self.df.set_index('index', inplace=True)
         self.update_data()
-
-        self.document.add_root((self.create()))
-        self.session = push_session(self.document)
+        self.source.on_change('data', self.update_from_table_change)
+        # XXX: I haven't had much luck running the server in this style
+        # self.document.add_root((self.create()))
+        # self.session = push_session(self.document)
```

Comment:
This was my attempt at running `buttons_server.py` which is in the main branch, not the datatables example. Are you able to get `buttons_server.py` to load successfully?

If I don't run it with `BOKEH_RESOURCES` then the following error is thrown when I attempt to load the page.

```
ERROR:tornado.application:Uncaught exception GET / (192.168.200.76)
HTTPServerRequest(protocol='http', host='192.168.200.161:5006', method='GET', uri='/', version='HTTP/1.1', remote_ip='192.168.200.76', headers={'Host': '192.168.200.161:5006', 'Upgrade-Insecure-Requests': '1', 'Accept-Language': 'en-US,en;q=0.8', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8', 'Connection': 'keep-alive', 'Cookie': 'username-192-168-200-161-8888=2|1:0|10:1471962824|29:username-192-168-200-161-8888|48:MDc5MTAwYzEtZTM0Ny00ODllLWE3YWItYmNkNDBiNzBhOTIw|e633d77ac704107e76705a4c0105d5e886a21435bab79f7d09f77f85a692ce0f', 'Dnt': '1', 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36', 'If-None-Match': '"f65cd282d20aaba016b7c6e425fac4dd1b066364"', 'Accept-Encoding': 'gzip, deflate, sdch'})
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/tornado/web.py", line 1445, in _execute
    result = yield result
  File "/usr/local/lib/python3.5/dist-packages/tornado/gen.py", line 1008, in run
    value = future.result()
  File "/usr/local/lib/python3.5/dist-packages/tornado/concurrent.py", line 232, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 3, in raise_exc_info
  File "/usr/local/lib/python3.5/dist-packages/tornado/gen.py", line 1017, in run
    yielded = self.gen.send(value)
  File "/usr/local/lib/python3.5/dist-packages/bokeh/server/views/doc_handler.py", line 30, in get
    page = server_html_page_for_session(session.id, self.application.resources(self.request),
  File "/usr/local/lib/python3.5/dist-packages/bokeh/server/tornado.py", line 325, in resources
    path_versioner=StaticHandler.append_version)
  File "/usr/local/lib/python3.5/dist-packages/bokeh/resources.py", line 236, in __init__
    raise ValueError("setting 'root_url' makes sense only when 'mode' is set to 'server'")
ValueError: setting 'root_url' makes sense only when 'mode' is set to 'server'
ERROR:tornado.access:500 GET / (192.168.200.76) 1.88ms
DEBUG:bokeh.server.tornado:[pid 6806] 1 clients connected
DEBUG:bokeh.server.tornado:[pid 6806]   / has 2 sessions with 1 unused
```